### ISSUE TITLE
Update CMake to newer version

### ISSUE THREAD

**Issue by rr-** [mxe/mxe#1260]
The feature I am missing was introduced in 3.1 and is `CMAKE_C_STANDARD` and `CMAKE_CXX_STANDARD`. Since `mxe` uses 3.0.2, I need to add silly workarounds for this.

Bump to 3.0.2 happened in December 2014; CMake is now at 3.5-rc2.

**Comment by starius** [comment]
I'm trying to build all packages with updated cmake:

```diff
diff --git a/src/cmake.mk b/src/cmake.mk
index f99868e..8fef047 100644
--- a/src/cmake.mk
+++ b/src/cmake.mk
@@ -3,8 +3,8 @@
 
 PKG             := cmake
 $(PKG)_IGNORE   :=
-$(PKG)_VERSION  := 3.0.2
-$(PKG)_CHECKSUM := 6b4ea61eadbbd9bec0ccb383c29d1f4496eacc121ef7acf37c7a24777805693e
+$(PKG)_VERSION  := 3.5.0
+$(PKG)_CHECKSUM := 92c83ad8a4fd6224cf6319a60b399854f55b38ebe9d297c942408b792b1a9efa
 $(PKG)_SUBDIR   := cmake-$($(PKG)_VERSION)
 $(PKG)_FILE     := cmake-$($(PKG)_VERSION).tar.gz
 $(PKG)_URL      := http://www.cmake.org/files/v$(call SHORT_PKG_VERSION,$(PKG))/$($(PKG)_FILE)
```

Can you check that the feature you need works after applying this patch, please?

**Comment by rr-** [comment]
I confirm that the patch lets me use the feature I need.

**Comment by starius** [comment]
Some packages are broken with new cmake: vmime, qtbase.

**Comment by starius** [comment]
CMake is updated as a part of the following pull request: https://github.com/mxe/mxe/pull/1279

**Comment by tonytheodore** [comment]
Updated in #1290. `vmime` was fixed in #1287 and `qtbase` no longer has issues (maybe 3.5.1 fixed it).
